Unit testing for log analyzer

alaakorda opened the pull request.

alaakorda committed:
chore: Add spring-boot-starter-test dependency and update pom.xml

alaakorda committed:
Remove unused code

alaakorda committed:
Merge branch 'main' into Unit-testing-for-logAnalyzer

alaakorda committed:
Update pom.xml and add unit test for influxDBController class

alaakorda committed:
Make test for queryData function

alaakorda committed:
edit test queryData function in exception case

alaakorda committed:
Refactor testQueryData function to handle exception case

alaakorda committed:
Make test for influxDbService functions

elgemmy commented on pom.xml:
```diff
@@ -69,6 +69,16 @@
+			<groupId>org.mockito</groupId>
+			<artifactId>mockito-core</artifactId>
+			<scope>test</scope>
+       </dependency>
```
Please fix the indentation for the added dependencies

elgemmy commented on src/test/java/com/service_health_monitor_portal/log_analyzer/influxDBControllerTest.java:
- Please aim to make your code self-explanatory, using comments only when necessary. Keep comments concise and expressive, avoiding specific details that might change in the future.
- Always keep an eye on the indentation of the code.
- It would enhance readability if you place single-line comments above the targeted line, rather than in front of it.
- Remove unnecessary comments, such as 'Test for writeData method'.

elgemmy reviewed (approved):
Good job! You put in a great effort!

efraimnabil committed:
Create Log Analyzer Service Test
- Update LogAnalyzerService.java class to create constructor and make the methods public
- Create LogAnalzerServiceTest Class
- Mock influxDBService and objectMapper
- InjectMocks into logAnalyzerService
- Setup to create instance from LogAnlyzerService beforEach
- Test ProcessLogLine
- Test getTimestamp
- Test isValidJson
